Update usage docs - add packer init (#104)
hashicorp/docker-hub-images: packer/README.md

```

#### Running a build:

The easiest way to run a command that references a configuration with one or more template files, is to mount a volume for the local workspace.

Running `packer init`
```shell
docker run \
    -v `pwd`:/workspace -w /workspace \
    -e PACKER_PLUGIN_PATH=/workspace/.packer.d/plugins \
    hashicorp/packer:latest \
    init .
```

~> **Note**: packer init is available from Packer v1.7.0 and later

The command will mount the working directory (`pwd`) to `workspace`, which is the working directory (`-w`) inside the container. 
Any plugin installed with `packer init` will be installed at the `PACKER_PLUGIN_PATH`. The `PACKER_PLUGIN_PATH` must be set to a path inside the volume mount so that plugins can become available at `packer build`. 

Running `packer build`
```shell
docker run \
    -v `pwd`:/workspace -w /workspace \
    -e PACKER_PLUGIN_PATH=/workspace/.packer.d/plugins \
    hashicorp/packer:latest \
    build .
```
##### Building old-legacy JSON templates

For old-legacy JSON, the build command must specify the template file(s).

```shell
docker run \
    -v `pwd`:/workspace -w /workspace \
    hashicorp/packer:latest \
    build template.json
```

> If plugins are necessary, we recommended to have the binaries inside a folder in the working directory and pass `-e PACKER_PLUGIN_PATH=/workspace/<wdir_plugin_path>` to the command above.

Reviewer: By plugins, do you mean third-party plugins not installed via Packer init? 

If so, what do you think about 
```suggestion
For the [manual installation](https://www.packer.io/docs/plugins#installing-plugins) of third-party plugins, we recommended that plugin binaries be placed a sub-directory under the working directory and adding `-e PACKER_PLUGIN_PATH=/workspace/<subdirectory_plugin_path>` to the command above.
```

Author: Yes. But at some point I caught myself thinking if on 1.8.0 JSON users would have to install every plugin manually. T I remembered that this won't be necessary with Megan's PoC for importing plugins into packer.

> For the [manual installation](https://www.packer.io/docs/plugins#installing-plugins) of third-party plugins, we recommended that plugin binaries be placed a sub-directory under the working directory and adding `-e PACKER_PLUGIN_PATH=/workspace/<subdirectory_plugin_path>` to the command above.

Author: ```suggestion
For the [manual installation](https://www.packer.io/docs/plugins#installing-plugins) of third-party plugins, we recommended that plugin binaries are placed under a sub-directory under the working directory. Add `-e PACKER_PLUGIN_PATH=/workspace/<subdirectory_plugin_path>` to the command above to tell Packer where the plugins are.
```
